public/app/features/dashboard-scene/saving/getDashboardChanges.ts

```diff
@@ -30,24 +35,40 @@ export function isEqual(a: VariableOption | undefined, b: VariableOption | undef
   return a === b || (a && b && a.selected === b.selected && deepEqual(a.text, b.text) && deepEqual(a.value, b.value));
 }
 
-// TODO[schema v2]
 export function getRawDashboardV2Changes(
   initial: DashboardV2Spec,
   changed: DashboardV2Spec,
   saveTimeRange?: boolean,
   saveVariables?: boolean,
   saveRefresh?: boolean
 ) {
+  const initialSaveModel = initial;
+  const changedSaveModel = changed;
+  const hasTimeChanged = getHasTimeChanged(changedSaveModel.timeSettings, initialSaveModel.timeSettings);
+  const hasVariableValueChanges = applyVariableChangesV2(changedSaveModel, initialSaveModel, saveVariables);
+  const hasRefreshChanged = changedSaveModel.timeSettings.autoRefresh !== initialSaveModel.timeSettings.autoRefresh;
+
+  if (!saveTimeRange) {
+    changedSaveModel.timeSettings.from = initialSaveModel.timeSettings.from;
+    changedSaveModel.timeSettings.to = initialSaveModel.timeSettings.to;
+  }
+
+  if (!saveRefresh) {
+    changedSaveModel.timeSettings.autoRefresh = initialSaveModel.timeSettings.autoRefresh;
+  }
+
+  const diff = jsonDiff(initialSaveModel, changedSaveModel);
+  const diffCount = Object.values(diff).reduce((acc, cur) => acc + cur.length, 0);
+
   return {
-    changedSaveModel: changed,
-    initialSaveModel: initial,
-    diffs: jsonDiff(initial, changed),
-    diffCount: 0,
-    hasChanges: false,
-    hasTimeChanges: false,
-    isNew: false,
```

Comment:
Is new, is something that lives in the metadata of the DashboardScene. It is not needed here